src/node_dotenv.h

```diff
@@ -32,6 +35,170 @@ class Dotenv {
 
 }  // namespace node
 
+namespace cppnv {
+
+struct VariablePosition {
+  ~VariablePosition() {
+    delete variable_str;
+  }
+
+  VariablePosition(int variable_start, int start_brace, int dollar_sign);
+  int variable_start;
+  int start_brace;
+  int dollar_sign;
+  int end_brace;
+  int variable_end;
+  std::string* variable_str;
+  bool closed = false;
+};
+class EnvStream {
+  size_t index_ = 0;
+  std::string* data_ = nullptr;
+  size_t length_;
+  bool is_good_;
+
+ public:
+  explicit EnvStream(std::string* data);
+  char get();
+  [[nodiscard]] bool good() const;
+  [[nodiscard]] bool eof() const;
+};
+struct EnvValue {
+  std::string* value;
+  bool is_parsing_variable = false;
+  std::vector<VariablePosition*>* interpolations;
+  int interpolation_index = 0;
+  bool quoted = false;
+  bool triple_quoted = false;
+  bool double_quoted = false;
+  bool triple_double_quoted = false;
+  bool implicit_double_quote = false;
+  bool back_tick_quoted = false;
+  int value_index = 0;
+  bool is_already_interpolated = false;
+  bool is_being_interpolated = false;
+  bool did_over_flow = false;
+  int back_slash_streak = 0;
+  int single_quote_streak = 0;
+  int double_quote_streak = 0;
+  std::string* own_buffer;
+
+
+  void clip_own_buffer(int length) const {
+    own_buffer->resize(length);
+  }
+
+  bool has_own_buffer() const {
+    return own_buffer != nullptr;
+  }
+
+  void set_own_buffer(std::string* buff) {
```

Comment:
Lack of documentation on these functions make it really hard to review/approve this PR